xds/server.go

```diff
@@ -338,13 +342,15 @@ func (s *GRPCServer) handleSecurityConfig(config *xdsclient.SecurityConfig, lw *
 	lw.xdsHI.SetRootCertProvider(rootProvider)
 	lw.xdsHI.SetIdentityCertProvider(identityProvider)
 	lw.xdsHI.SetRequireClientCert(config.RequireClientCert)
+	lw.providerMu.Unlock()
```

Comment:
You can move this above the `xdsHI` parts, right?

Reply:
The `HandshakeInfo` type does use a mutex in its methods. In a very unlikely scenario, we could have two Listener updates come right after one another, and having the lock for the whole duration will make sure that the contents of `xdsHI` at the end of it all will be sane.